sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -31,9 +31,9 @@
     protected AbstractQueryGenerator() {
     }
 
-    private String generateQueryParameter(@NonNull String subject) {
+    private String generateQueryParameter(@NonNull String subject, int counter) {
         // user.name, user['name'] or user["first name"] are not valid sql parameter identifiers.
-        return subject.replaceAll("[^a-zA-Z\\d]", "_") + UUID.randomUUID().toString().replaceAll("-", "_");
+        return subject.replaceAll("[^a-zA-Z\\d]", "_") + counter;
```

Comment:
Can you please elaborate on this logic

Reply:
@simplynaveen20   In Spring, you can have duplicate parameters in query which are of the same name - for example this query - `Iterable<Project> findByNameAndCreatorOrNameAndCreator(String name, String creator, String name2, String creator2);`
This would get converted to this - 
`select * from c where c.name = @name and c.creator = @creator OR c.name = @name and c.creator = @creator`
This cause problems when running the query - as Gateway throws exception saying 400 bad request. 
To avoid this issue, when we are generating the query params, we were appending random UUID to have different query param names for same fields. 
However, that was resulting in a different query text for the same query if executed twice in the same JVM. 
To solve the query text issue, now I am using a counter - which gets initialized when generating a query text. That way, same queries (conceptually) now will have same query text. Which is what we want.

Reply:
This was the issue that we fixed earlier for duplicate query params in the same query - https://github.com/Azure/azure-sdk-for-java/issues/17961
And this was the fix that we did last time - https://github.com/Azure/azure-sdk-for-java/pull/17962

That's why it is important to have a counter. However, counter is not random so it will have the same query text every time the same query is executed. Counter is just added to differentiate between duplicate query param names.